### DISCUSSION TITLE
Grid and Angle Snapping

### DISCUSSION THREAD

**Discussion by andybak** [icosa-gallery/open-brush#175] [New and Experimental Features]
Grab a build here and try it out: https://github.com/IxxyXR/open-brush/actions/runs/1286742960#artifacts
Note - you need to be logged in to Github to download. If that's a pain for anyone let me know and i'll send you a build
@AncientWorlds has been testing and we've got the following feedback noted:
1. Snapping only works when you grab a selection (Duplicating without grabbing isn't currently working)
2. The icon on the button doesn't always appear
3. Angle snap is a bit unintuitive in how it works - I'm going to compare with how Medium does it
4.  Snapping must be turned on in the Labs panel
         (At this time the snapping settings overlay the original panel on first launch,  drag the panel to a new position to use
         The panel has buttons to control angle snap and the amount of grid spacing)
5.  To use snapping :: activate select tool > Grab > drag > Rotate Controller
         dragging snaps to the next grid point
         rotating the controller snaps the angle in increments
6. Snapping duplication behaves differently depending on the selection method used.
         Method 1:  hover over a selection and press the duplicate button.
          Method 2:press the duplicate button whilst grabbing a selection
         Note:  Duplication Method two will copy the object in place, Method 1 will offset the object

**Comment by Gemcrafter** [comment]
Note:  need to look at mirror compatibility at some point
